lib/ansible/module_utils/basic.py

```diff
@@ -1099,9 +1099,7 @@ def _apply_operation_to_mode(user, operator, mode_to_apply, current_mode):
         return new_mode
 
     @staticmethod
-    def _get_octal_mode_from_symbolic_perms(path_stat, user, perms, use_umask):
-        prev_mode = stat.S_IMODE(path_stat.st_mode)
-
+    def _get_octal_mode_from_symbolic_perms(path_stat, prev_mode, user, perms, use_umask):
```

Comment:
Add the new option at the end, and fall back to the value from the path_stat if it isn't passed:

```suggestion
    def _get_octal_mode_from_symbolic_perms(path_stat, user, perms, use_umask, prev_mode=None):
        if prev_mode is None:
            prev_mode = stat.S_IMODE(path_stat.st_mode)
```

Reply:
Sorry, I didn't see your comment earlier.  I've added this change, though I'll point out that as far as I can tell this code is only called from the one location.  Unless some third party modules are calling it?  Though I would assume that the "_" prefix means "for internal use only"?  While normally I would agree that adding it to the end and allowing calling without the new parameter is the best plan, if it is only used in one other place and that place always supplies the new argument, in this case the "if prev_mode is None" results in dead code.

Thoughts?

Reply:
@lilatomic No problem. Generally when I'm reviewing I will ask people to remove unnecessary cosmetic changes, and since there was no reason to change argument order this was no different.

> While normally I would agree that adding it to the end and allowing calling without the new parameter is the best plan, if it is only used in one other place and that place always supplies the new argument, in this case the "if prev_mode is None" results in dead code.

It could be added to the end as a required arg - but since this is in module_utils my knee jerk reaction was to keep it backwards compatible. Most modules are probably calling this  via `_symbolic_mode_to_octal` (also 'private'). You could add a test in test/units/module_utils/basic/test__symbolic_mode_to_octal.py for backwards compatibility if you wanted to make sure this code is exercised.